pkg/azuredisk/controllerserver.go

```diff
@@ -194,18 +198,6 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		}
 	}
 
-	if maxShares < 2 {
```

Comment:
I think the reasonable logic is like this:
if maxShares >=2 {
  allow muti-node read/write for raw block device
} 
always allow single-node read/write no matter it's shared disk or not.

Reply:
I wanted to keep the volume capabilities check separate from maxshares values as we per [CSI documentation](https://kubernetes-csi.github.io/docs/raw-block.html?search=ValidateVolumeCapabilities) we need to validate volume capabilities in the following calls where we don't necessarily want to parse the maxshares values: 
CreateVolume() (multiple capabilities)
ControllerPublishVolume()
ValidateVolumeCapabilities() (multiple capabilities)
GetCapacity() (see below)
NodeStageVolume()
NodePublishVolume()

Do you think we should allow multi-node read/write for file system if the maxshares is < 2? That's going to be the only difference in the current and suggested implementation. 
We could replace the `IsValidVolumeCapabilities` with this logic here but then again, I think it makes it hard to maintain and keep the validation logic consistent.

Reply:
This PR actually changed the logic: it allows RWX with raw block device now (no matter it's shared disk or not), original design is that only shared disk would supports RWX with raw block device

Reply:
If the disk is not shared, does allowing or not allowing RWX make any difference since sharing is already disabled? Alternatively, do you think we should be relying on maxshares for verifying volume capabilities in all the following functions?: 
CreateVolume() (multiple capabilities)
ControllerPublishVolume()
ValidateVolumeCapabilities() (multiple capabilities)
GetCapacity() (see below)
NodeStageVolume()
NodePublishVolume()

Reply:
if disk is not shared, we should not allow RWX. I think we need to keep original design, only shared disk would support RWX, RWO with raw block device, and yes, this driver should also rely on maxshares, and if `maxshares` is not provided by user, then it's not shared disk by default.

Reply:
I have added maxshares parameter to the verification logic.